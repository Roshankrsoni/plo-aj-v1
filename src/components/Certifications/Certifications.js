import "./Certifications.css";

function Certifications(props) {
  return (
    <div
      className={
        "show-on-scroll col-md-7 page-1 text-left section certifications " +
        props.class
      }
    >
      <div className="section-title grey4 h2 bold">Certifications</div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Development
        </div>

        {props.dbData.certifications.development.map((ele, key) => {
          return <div className="subsection-data" key={key}>
            <span className="h3 grey1 bold">{ele.name}</span>
            <div className="red default-text mt-1">
              <a href={ele.urlToCertificate} target="_Blank" rel="noopener noreferrer">
                View LinkedIn profile
                </a>
            </div>
            <div className="mt-2 grey-1 body-text">{ele.summary}</div>
          </div>

        })}
      </div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Design
        </div>
        {props.dbData.certifications.design.map((ele, key) => {
          return <div className="subsection-data" key={key}>
            <span className="h3 grey1 bold">{ele.name}</span>
            <div className="red default-text mt-1">
              <a href={ele.urlToCertificate} target="_Blank" rel="noopener noreferrer">
                View LinkedIn profile
                </a>
            </div>
            <div className="mt-2 grey-1 body-text">{ele.summary}</div>
          </div>

        })}
      </div>
    </div>
  );
}

export default Certifications;
