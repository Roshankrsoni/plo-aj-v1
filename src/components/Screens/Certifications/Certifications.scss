.certifications-content {
  .summary-text {
    padding-top: 0.5rem;
  }
}
