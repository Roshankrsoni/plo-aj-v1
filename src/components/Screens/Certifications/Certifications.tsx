import React from "react";
import { useGlobalContext } from "../../../context/GlobalContext";
import { CertificateDetails } from "../../../types/profileDataTypes";
import SectionVisibilityHOC from "../../Organisms/SectionVisibilityHOC/SectionVisibilityHOC";
import T from "../../../translations/en_IN";
import constants from "../../../helpers/constants";

const Certifications = () => {
    const { profileData } = useGlobalContext();
    const { certifications } = profileData;
    const { development, design } = certifications;
    return (
        <>
            <div className="section-title h2 bold">{T.CERTIFICATIONS}</div>
            <div className="subsection">
                <div className="subsection-title uppercase body-text grey5 letterspacing-1">{T.DEVELOPMENT}</div>

                {development.map((certificateDetail: CertificateDetails, key: number) => {
                    return (
                        <div className="subsection-data" key={key}>
                            <span className="h3 grey1 bold" data-testid="name">
                                {certificateDetail.name}
                            </span>
                            <div className="primary-color default-text mt-1">
                                <a href={certificateDetail.urlToCertificate} target="_Blank" rel="noopener noreferrer">
                                    View LinkedIn profile
                                </a>
                            </div>
                            <div className="mt-2 grey-1 body-text">{certificateDetail.summary}</div>
                        </div>
                    );
                })}
            </div>
            <div className="subsection">
                <div className="subsection-title uppercase body-text grey5 letterspacing-1">{T.DESIGN}</div>
                {design.map((certificateDetail: CertificateDetails, key: number) => {
                    return (
                        <div className="subsection-data" key={key}>
                            <span className="h3 grey1 bold">{certificateDetail.name}</span>
                            <div className="primary-color default-text mt-1">
                                <a href={certificateDetail.urlToCertificate} target="_Blank" rel="noopener noreferrer">
                                    {T.VIEW_LINKEDIN_PROFILE}
                                </a>
                            </div>
                            <div className="mt-2 grey-1 body-text">{certificateDetail.summary}</div>
                        </div>
                    );
                })}
            </div>
        </>
    );
};

export default SectionVisibilityHOC(Certifications, constants.classNames.CERTIFICATIONS);
