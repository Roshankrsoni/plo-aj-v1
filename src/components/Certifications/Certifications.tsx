import React from "react";
import "./Certifications.css";
import { useGlobalContext } from "../../context/GlobalContext";
import { CertificateDetails } from "../../Types/profileDataTypes";

const Certifications = () => {
  const { profileData } = useGlobalContext();
  const { certifications } = profileData;
  const { development, design } = certifications;
  return (
    <div className={"col-md-7 page-1 text-left section certifications"}>
      <div className="section-title h2 bold">Certifications</div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Development
        </div>

        {development.map(
          (certificateDetail: CertificateDetails, key: number) => {
            return (
              <div className="subsection-data" key={key}>
                <span className="h3 grey1 bold" data-testid="name">
                  {certificateDetail.name}
                </span>
                <div className="red default-text mt-1">
                  <a
                    href={certificateDetail.urlToCertificate}
                    target="_Blank"
                    rel="noopener noreferrer"
                  >
                    View LinkedIn profile
                  </a>
                </div>
                <div className="mt-2 grey-1 body-text">
                  {certificateDetail.summary}
                </div>
              </div>
            );
          }
        )}
      </div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Design
        </div>
        {design.map((certificateDetail: CertificateDetails, key: number) => {
          return (
            <div className="subsection-data" key={key}>
              <span className="h3 grey1 bold">{certificateDetail.name}</span>
              <div className="red default-text mt-1">
                <a
                  href={certificateDetail.urlToCertificate}
                  target="_Blank"
                  rel="noopener noreferrer"
                >
                  View LinkedIn profile
                </a>
              </div>
              <div className="mt-2 grey-1 body-text">
                {certificateDetail.summary}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Certifications;

export { Certifications };
