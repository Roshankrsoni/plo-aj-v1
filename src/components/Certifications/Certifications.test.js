import React from 'react';
import { shallow } from 'enzyme';
import Certifications from './Certifications';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Certifications dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});