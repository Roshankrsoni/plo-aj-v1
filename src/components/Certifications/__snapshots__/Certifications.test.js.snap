// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section certifications undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Certifications
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-title uppercase body-text grey3 letterspacing-1"
    >
      Development
    </div>
    <div
      className="subsection-data"
      key="0"
    >
      <span
        className="h3 grey1 bold"
      >
        React Js Hackerrank Certification
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        React Js - Basic Hackerrank certifcation.
      </div>
    </div>
    <div
      className="subsection-data"
      key="1"
    >
      <span
        className="h3 grey1 bold"
      >
        Microsoft Technology Associate - Web Development Fundamentals
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Started my journey as a UI developer in my college days where I opted for this certification which involved learning HTML, CSS, JavaScript and .Net Fundamentals.
      </div>
    </div>
    <div
      className="subsection-data"
      key="2"
    >
      <span
        className="h3 grey1 bold"
      >
        HTML5: Structure, Syntax, and Semantics
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com
      </div>
    </div>
    <div
      className="subsection-data"
      key="3"
    >
      <span
        className="h3 grey1 bold"
      >
        Learning Git and GitHub
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com
      </div>
    </div>
    <div
      className="subsection-data"
      key="4"
    >
      <span
        className="h3 grey1 bold"
      >
        Shaping up with Angular.js
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com
      </div>
    </div>
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-title uppercase body-text grey3 letterspacing-1"
    >
      Design
    </div>
    <div
      className="subsection-data"
      key="0"
    >
      <span
        className="h3 grey1 bold"
      >
        Drawing Vector Graphics: Iconography
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com - Deke McClelland
      </div>
    </div>
    <div
      className="subsection-data"
      key="1"
    >
      <span
        className="h3 grey1 bold"
      >
        Illustrator CC 2017 One-on-One Fundamentals
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com - Deke McClelland
      </div>
    </div>
    <div
      className="subsection-data"
      key="2"
    >
      <span
        className="h3 grey1 bold"
      >
        Creating a Design System with Sketch
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        Lynda.com - Deke McClelland
      </div>
    </div>
    <div
      className="subsection-data"
      key="3"
    >
      <span
        className="h3 grey1 bold"
      >
        Learning Graphic Design: Layouts, Presentations, Cropping Photographs, Things Every Designer Should Know
      </span>
      <div
        className="red default-text mt-1"
      >
        <a
          href="https://www.linkedin.com/in/anoop-jadhav-44528258/"
          rel="noopener noreferrer"
          target="_Blank"
        >
          View LinkedIn profile
        </a>
      </div>
      <div
        className="mt-2 grey-1 body-text"
      >
        John McWade
      </div>
    </div>
  </div>
</div>
`;
